test/thirdparty.test.ts

```diff
@@ -0,0 +1,227 @@
+import { describe, test } from "mocha";
+import { getAccounts } from "@tableland/local";
+import {
+  D1Orm,
+  DataTypes,
+  Model,
+  GenerateQuery,
+  QueryType,
+  type Infer,
+} from "d1-orm";
+import sql, { FormatConfig } from "@databases/sql";
+import { escapeSQLiteIdentifier } from "@databases/escape-identifier";
+import { NonceManager } from "@ethersproject/experimental";
+import { getDefaultProvider, overrideDefaults } from "../src/helpers/index.js";
+import { Database } from "../src/index.js";
+
+overrideDefaults("localhost", {
+  contractAddress: "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
+});
+
+describe("thirdparty", function () {
+  // Note that we're using the second account here
+  const [, wallet] = getAccounts();
+  const provider = getDefaultProvider("http://127.0.0.1:8545");
+  // const signer = wallet.connect(provider);
+  const baseSigner = wallet.connect(provider);
+  // Also demonstrates the nonce manager usage
+  const signer = new NonceManager(baseSigner);
+  const db = new Database({ signer });
+
+  describe("d1-orm", function () {
+    const orm = new D1Orm(db);
+
+    // We'll define our core model up here and use it in tests below
+    const users = new Model(
+      {
+        D1Orm: orm,
+        tableName: "users",
+        primaryKeys: "id",
+        uniqueKeys: [["email"]],
+      },
+      {
+        id: {
+          type: DataTypes.INTEGER,
+          notNull: true,
+        },
+        name: {
+          type: DataTypes.STRING,
+          notNull: true,
+          // See https://github.com/Interactions-as-a-Service/d1-orm/issues/60
+          // defaultValue: "John Doe",
+        },
+        email: {
+          type: DataTypes.STRING,
+        },
+      }
+    );
+    type User = Infer<typeof users>;
+
+    this.beforeAll(async function () {
+      this.timeout("10s");
+
+      const create = await users.CreateTable({
+        strategy: "default",
+      });
+      await create.meta.txn.wait();
+
+      // TODO: Find a nicer way to deal with this...
+      (users.tableName as any) = create.meta.txn.name;
+    });
+
+    test("where a basic model is used to create data", async function () {
+      await users.InsertOne({
+        name: "Bobby Tables",
+        email: "bobby-tab@gmail.com",
+      });
+
+      const [result] = await users.InsertMany([
+        {
+          name: "Bobby Tables",
+          email: "bob-tables@gmail.com",
+        },
+        {
+          name: "Jane Tables",
+          email: "janet@gmail.com",
+        },
+      ]);
+
+      await result.meta.txn.wait();
+
+      const { results } = await users.All({
+        where: { name: "Bobby Tables" },
+        limit: 1,
+        offset: 0,
+        orderBy: ["id"],
+      });
+
+      deepStrictEqual(results, [
+        {
+          name: "Bobby Tables",
+          id: 1,
+          email: "bobby-tab@gmail.com",
+        },
+      ]);
+    });
+
+    test("basic query building works well to then query the data", async function () {
```

Comment:
This query builder pattern is kinda nice. There are others that also work with D1, but this one is the most advanced IMO.